src/Symfony/Component/Security/Http/Event/LogoutOnChangeEvent.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Event;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * @author Hamza Amrouche <hamza.simperfit@gmail.com>
+ */
+class LogoutOnChangeEvent extends Event
+{
+    private $token;
+    private $user;
+
+    public function __construct(TokenInterface $token, UserInterface $user)
+    {
+        $this->token = $token;
+        $this->user = $user;
+    }
+
+    public function getToken(): TokenInterface
+    {
+        return $this->token;
+    }
+
+    public function getUser(): UserInterface
```

Comment:
Can you add documentation on which user and which token to expect here? Reading your code I realized I wasn't able to quickly determine which I'd be receiving

Reply:
done